Invalid DDL for Oracle lock table

### Play Version (2.5.x / etc)

2.5.6
### API (Scala / Java / Neither / Both)

Scala
### Operating System (Ubuntu 15.10 / MacOS 10.10 / Windows 10)

Darwin jasons-macbook 16.0.0 Darwin Kernel Version 16.0.0: Mon Aug 29 17:56:20 PDT 2016; root:xnu-3789.1.32~3/RELEASE_X86_64 x86_64
### JDK (Oracle 1.8.0_72, OpenJDK 1.8.x, Azul Zing)

java version "1.8.0_102"
Java(TM) SE Runtime Environment (build 1.8.0_102-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.102-b14, mixed mode)
### Library Dependencies

Oracle database OS: OEL 6.6 ( 2.6.39-400.17.1.el6uek) x86_64
Oracle database version: 12.1.0.1.0 patch 17027533
Oracle JDBC driver: oracle:ojdbc7:12.1.0.2.0 (this a locally published copy)
### Expected Behavior

Please describe the expected behavior of the issue, starting from the first action.
1. Add `"oracle" % "ojdbc7" % "12.1.0.2.0"` `PlayImport.jdbc` and `PlayImport.evolutions` to the `libraryDependencies`.
2. Configure `db.default` datasource to point to the Oracle database and use the Oracle JDBC driver.
3. Configure `play.evolutions` with `useLocks` and `autoApply` enabled (set to `true`).
4. Create a database evolution
5. Run the Play application
6. Make a request to a controller
7. `play_evolutions_lock` table is created
### Actual Behavior

Play starts as normal

```
--- (Running the application, auto-reloading is enabled) ---

[info] p.c.s.NettyServer - Listening for HTTPS on /0:0:0:0:0:0:0:0:9000

(Server started, use Ctrl+D to stop and go back to the console...)
```

Play returns 500 Internal Server error

```
[error] application - 

! @71mk1megm - Internal server error, for (GET) [/] ->

play.api.UnexpectedException: Unexpected exception[SQLSyntaxErrorException: ORA-00904: : invalid identifier
]
    at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1$$anonfun$1.apply(DevServerStart.scala:180)
    at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1$$anonfun$1.apply(DevServerStart.scala:131)
    at scala.Option.map(Option.scala:146)
    at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1.apply(DevServerStart.scala:131)
    at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1.apply(DevServerStart.scala:129)
    at scala.util.Success.flatMap(Try.scala:231)
    at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1.apply(DevServerStart.scala:129)
    at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1.apply(DevServerStart.scala:121)
    at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
    at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
Caused by: java.sql.SQLSyntaxErrorException: ORA-00904: : invalid identifier

    at oracle.jdbc.driver.T4CTTIoer.processError(T4CTTIoer.java:450)
    at oracle.jdbc.driver.T4CTTIoer.processError(T4CTTIoer.java:399)
    at oracle.jdbc.driver.T4C8Oall.processError(T4C8Oall.java:1059)
    at oracle.jdbc.driver.T4CTTIfun.receive(T4CTTIfun.java:522)
    at oracle.jdbc.driver.T4CTTIfun.doRPC(T4CTTIfun.java:257)
    at oracle.jdbc.driver.T4C8Oall.doOALL(T4C8Oall.java:587)
    at oracle.jdbc.driver.T4CStatement.doOall8(T4CStatement.java:210)
    at oracle.jdbc.driver.T4CStatement.doOall8(T4CStatement.java:30)
    at oracle.jdbc.driver.T4CStatement.executeForRows(T4CStatement.java:931)
    at oracle.jdbc.driver.OracleStatement.doExecuteWithTimeout(OracleStatement.java:1150)
```

The problem is in `play.api.db.evolutions.ApplicationEvolutions#CreatePlayEvolutionsLockOracleSql` which has the following DDL template:

```
    """
      CREATE TABLE ${schema}play_evolutions_lock (
        lock Number(10,0) Not Null Enable,
        CONSTRAINT play_evolutions_lock_pk PRIMARY KEY (lock)
      )
    """
```

`lock` is a [Oracle SQL Reserved Word](https://docs.oracle.com/database/121/SQLRF/ap_keywd001.htm#SQLRF55621).

It could be fixed by quoting the column identifier like in the MySQL DDL templates but why not just avoid the hassle and use `lock_id` or something?

I will hopefully create a PR for this in a day or two but in the meantime I guess I will just have to disable the locking.

--- a/framework/src/play-jdbc-evolutions/src/main/scala/play/api/db/evolutions/ApplicationEvolutions.scala
+++ b/framework/src/play-jdbc-evolutions/src/main/scala/play/api/db/evolutions/ApplicationEvolutions.scala
@@ -107,7 +107,7 @@ class ApplicationEvolutions @Inject() (
     import ApplicationEvolutions._
     val (selectScript, createScript, insertScript) = url match {
       case OracleJdbcUrl() =>
-        (SelectPlayEvolutionsLockSql, CreatePlayEvolutionsLockOracleSql, InsertIntoPlayEvolutionsLockSql)
+        (SelectPlayEvolutionsLockOracleSql, CreatePlayEvolutionsLockOracleSql, InsertIntoPlayEvolutionsLockOracleSql)
       case MysqlJdbcUrl(_) =>
         (SelectPlayEvolutionsLockMysqlSql, CreatePlayEvolutionsLockMysqlSql, InsertIntoPlayEvolutionsLockMysqlSql)
       case _ =>
@@ -128,6 +128,7 @@ class ApplicationEvolutions @Inject() (
     import ApplicationEvolutions._
     val lockScripts = url match {
       case MysqlJdbcUrl(_) => lockPlayEvolutionsLockMysqlSqls
+      case OracleJdbcUrl() => lockPlayEvolutionsLockOracleSqls
       case _ => lockPlayEvolutionsLockSqls
     }
     try {
@@ -171,6 +172,11 @@ private object ApplicationEvolutions {
       select `lock` from ${schema}play_evolutions_lock
     """
 
+  val SelectPlayEvolutionsLockOracleSql =
+    """
+      select "lock" from ${schema}play_evolutions_lock
+    """
+
   val CreatePlayEvolutionsLockSql =
     """
       create table ${schema}play_evolutions_lock (
@@ -188,8 +194,8 @@ private object ApplicationEvolutions {
   val CreatePlayEvolutionsLockOracleSql =
     """
       CREATE TABLE ${schema}play_evolutions_lock (
-        lock Number(10,0) Not Null Enable,
-        CONSTRAINT play_evolutions_lock_pk PRIMARY KEY (lock)
+        "lock" Number(10,0) Not Null Enable,
+        CONSTRAINT play_evolutions_lock_pk PRIMARY KEY ("lock")
       )
     """
 
@@ -201,6 +207,11 @@ private object ApplicationEvolutions {
   val InsertIntoPlayEvolutionsLockMysqlSql =
     """
       insert into ${schema}play_evolutions_lock (`lock`) values (1)
+    """
+
+  val InsertIntoPlayEvolutionsLockOracleSql =
+    """
+      insert into ${schema}play_evolutions_lock ("lock") values (1)
     """
 
   val lockPlayEvolutionsLockSqls =
@@ -217,6 +228,13 @@ private object ApplicationEvolutions {
       """,
       """
         select `lock` from ${schema}play_evolutions_lock where `lock` = 1 for update
+      """
+    )
+
+  val lockPlayEvolutionsLockOracleSqls =
+    List(
+      """
+        select "lock" from ${schema}play_evolutions_lock where "lock" = 1 for update nowait
       """
     )
 }

